File: Backend2/Model/Credentials/Address/AddressDto.cs
using System.ComponentModel.DataAnnotations;
using Fictichos.Constructora.Model;

namespace Fictichos.Constructora.Dto;
public record AddressDto : DtoBase
{
    public string? Street { get; set; }
    public string? Number { get; set; }
    public string? Colony { get; set; }
    public string? PostalCode { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? Country { get; set; }

    public AddressDto(Address data)
    {
        Street = data.Street;
        Number = data.Number;
        Colony = data.Colony;
        PostalCode = data.PostalCode;
        City = data.PostalCode;
        State = data.State;
        Country = data.Country;
    }
    public AddressDto() { }
}

public record SlimAddressDto
{
    public string? Street { get; set; }
    public string? Number { get; set; }
    public string? Colony { get; set; }
    public string? PostalCode { get; set; }

    public SlimAddressDto(Address data)
    {
        Street = data.Street;
        Number = data.Number;
        Colony = data.Colony;
        PostalCode = data.PostalCode;
    }
}

File: gambit/src/models/Response/Contact/AddressDto.ts
class AddressDto
{
  Street?: string;
  Number?: string;
  Colony?: string;
  PostalCode?: string;
  City?: string;
  State?: string;
  Country?: string;
  Coordinates?: Coordinates;

  constructor(street?: string, number?: string, colony?: string, postalCode?: string, city?: string, state?: string, country?: string, coordinates?: Coordinates)
  {
    this.Street = street;
    this.Number = number;
    this.Colony = colony;
    this.PostalCode = postalCode;
    this.City = city;
    this.State = state;
    this.Country = country;
    this.Coordinates = coordinates;
  }
}

class Coordinates
{
  Latitude: string;
  Longitude: string;

  constructor(lat: string, long: string)
  {
    this.Latitude = lat;
    this.Longitude = long;
  }
}

export default AddressDto;